### ISSUE TITLE
SA1118 Cannot be suppressed in attributes

### ISSUE THREAD

**Issue by simontherry** [DotNetAnalyzers/StyleCopAnalyzers#3676]
Hi,

I have an issue where I want to use an attribute `RequiredIfTypeIsAttribute` (a custom attribute for our project) where one of the parameter is a long line which upon code cleanup gets rewritten to multiple lines:

```
        [RequiredIfTypeIs(
            nameof(Type),
#pragma warning disable SA1118
            new[]
            {
                MyEnum.Value1,
                MyEnum.Value2,
                MyEnum.Value3,
                MyEnum.Value4,
                MyEnum.Value5,
                MyEnum.Value6,
                MyEnum.Value7,
                MyEnum.Value8,
                MyEnum.Value9,
            })]
#pragma warning restore SA1118
        public List<Guid>? Ids { get; set; }
```

I had to disable SA1118 on this line because I cannot instantiate this array using a const. How can I fix this?

**Comment by bjornhellander** [comment]
Hi! The suppression in your code snippet works fine for me. Suppression is handled by the compiler, so if it doesn't work for you then file an issue here: https://github.com/dotnet/roslyn.

**Comment by simontherry** [comment]
Yes, the suppression works fine, but I wanted to avoid the suppression, i.e. StyleCop should allow and not report SA1118 in this scenario because there is no alternative than the suppression.

**Comment by sharwell** [comment]
Is this not a duplicate of #2916?

**Comment by bjornhellander** [comment]
@sharwell Multi-line array initializers (and other exceptions) are allowed as e.g. method call arguments, but currently not in attribute arguments.

**Comment by sharwell** [comment]
Weird. I can't find a reason for that distinction.
